### ISSUE TITLE
Meal types and meal courses

### ISSUE THREAD

**Issue by ddooley** [FoodOntology/foodon#236]
As a result of the ICBO 2022 FoodOn food processing hackathon on Sept 25, discussion arose on formalizing meal types as a kind of food consumption process, and courses/dishes.

<img width="1026" alt="image" src="https://user-images.githubusercontent.com/4000582/195952171-394e221e-93d9-4b54-8015-619ce92a126c.png">

Feedback appreciated!

**Comment by ddooley** [comment]
P.s. there is resistance to creating classes of [culture] x breakfast food as precomposed terms.  Even at a regional level application of breakfast food may entail a different set of foods, which are not accepted by all people in that region.  Same with other meal category foods.

**Comment by cmrn-rhi** [comment]
JFO Meeting (2022-10-27) Follow-up: I think dishes are food preparations that can be "slotted" into "meal courses" (which are 1 or more dishes as we discussed in meeting). As we were closing  there were questions regarding "dessert course" vs "dessert dish" and I just want to capture my thoughts on it for next time: I think we should keep dessert course and it's relativity to other courses, but that we shouldn't have "dessert dish" as that doesn't seem like a useful shadow term - perhaps instead "confectionary dish" (or something similar) which would be an option that could be "slotted" into "dessert course" when applicable.

**Comment by maweber-bia** [comment]
Just need to define what a "confectionary dish" is ;-) but I agree it is a kind of preparation that can be slotted into a meal course  as discussed during the meeting

**Comment by cmrn-rhi** [comment]
> Just need to define what a "confectionary dish" is ;-) but I agree it is a kind of preparation that can be slotted into a meal course as discussed during the meeting

I agree! And I lean towards going off the definition for "confectionary food product" we discussed in #222

**Additionally:** It might be worth considering (in parallel to all this meal stuff) adding things like "dessert", "breakfast", "lunch", etc. as label claim classes because it's not unusual to see a "breakfast bar" or "kids lunch food pack" where people apply these labels but there's actually no guarantee of at what meal consumption occurs.

**Comment by ddooley** [comment]
Updated diagram from FoodOn curation call Oct 27:
<img width="957" alt="image" src="https://user-images.githubusercontent.com/4000582/198510978-ab985340-8312-492e-ae5f-eb887e3ea55d.png">
I threw in "Western Diet Breakfast Dish" and "Korean breakfast dish" just for illustration.  Not sold on actually adding those terms to foodon - its a combination bomb regarding countries, cultures, and meal types.

**Comment by cmrn-rhi** [comment]
Yeah, I'm not sold on adding "Western Diet Breakfast Dish" and "Korean breakfast dish" either. I feel like "Western"/"Korean" are a "style" (e.g. like "italian-style" currently in FoodOn) while "breakfast is just a label claim (or something similar) - both of which could be applied to a "dish" when needed but shouldn't be precomposed unless in high demand.

**Comment by RobertDanhi** [comment]
> 

Korean is a geographic location and can be defined yet Western is usually anything "non-Asian" hence not useful here. Also consider where Asian is 40+ countries and such we cannot use "asian" either.

**Comment by RobertDanhi** [comment]
>western diet breakfast dish
> > 
> 
> Korean is a geographic location and can be defined yet Western is usually anything "non-Asian" hence not useful here. Also consider where Asian is 40+ countries and such we cannot use "asian" either.

In regards to Asian, I suggest we plan for the future and have each country. Yet for top level grouping we can use the common Southeast Asian (then dive deeper into Thailand, Malaysia, Singapore, Indonesia, Cambodia, Los/Myanmar...), Northeast Asia (China, Korean, Japan, Taiwan, ...), South Asian (India, Sri Lanka, Pakistan, ...)

**Comment by RobertDanhi** [comment]
[Damion's edit - I dropped repeat of diagram for brevity]
I also posted this as separate issue, (just learning protocol in this platform/forum - A very common name for a time specific meal period is "supper", usually a meal occasion after dinner or instead of dinner after 10pm/ 22:00, then if folks go for some food much later after midnight (such as after bars/dancing/drinking) it can still be called supper of often referred to as "midnight snack" (this within Northeast and Southeast Asian context since in much or Europe such as Spain a majority of full service/fine dinning restaurants don't even open for dinner until 9pm)

**Comment by RobertDanhi** [comment]
[Damion's edit - I dropped repeat of diagram for brevity]

Canape is used in certain settings such as formal party, reception, event as well.

**Comment by RobertDanhi** [comment]
[Damion's edit - I dropped repeat of diagram for brevity]

Dinner is not the main meal in some parts of the world, I believe in Spain, lunch is considered more of the main meal.

**Comment by ddooley** [comment]
Hi Robert,
It is possible to set up a separate issue for each term (as you did with https://github.com/FoodOntology/foodon/issues/266 ), but I think we can at least handle preliminary discussion on this thread for what we want to capture for named things and their synonyms and semantics!

**Comment by ddooley** [comment]
Robert, here's a short intro about the choice between "pre-composing", i.e. creating a single term for a category, like "korean breakfast dish", versus having a database express such a thing expressed as a 2 variable structure, e.g. "korean" x "breakfast dish", or more generally, "country" x "meal dish" (say, as a few columns of a database "dish type" table).  Above I said "I threw in "Western Diet Breakfast Dish" and "Korean breakfast dish" just for illustration" - I meant that as a thought exercise about where those items would hypothetically fit. I don't think we do want to pre-compose them.

In FoodOn we do have a list of countries to draw from (currently OBO's GAZETTEER ontology) for use in that kind of "country" x "meal dish" expression.

The question about "western diet" - we actually had a discussion about this in a past JFOW call and came up with the ONS term "[globalized diet](http://purl.obolibrary.org/obo/ONS_1000065)" with synonym "western diet":  It has a long definition.  We can't really make a logical definition, but to me it seems useful to have this term as a kind of keyword that we can associate with studies that purport to be about western diet.  

That same question - do we need this term, and for what purpose - can be asked of any of the dish terms, and as well whether there is a best primary term in the case where a number of terms seem synonymous. If we find variations at work - dinner means this in one (cultural) context, and that in another, then we create a parent term that encompasses both senses of a term if that is possible, and then qualify the underlying terms by their context.  This can get tricky, and sometimes we have to look at kinds of synonymy - broad, narrow, related, exact - to see if there's a solution.  E.g. maybe

main meal:
  - related synonym: dinner
  - related synonym: lunch

This exposes the challenge of mapping natural language terms to ontology ones.

What you said about canape suggests that it is an exact synonym for snack, unless what a "canape" references is a (mysteriously defined) subset of snacks?  If canape is simply used in the context of certain events, but still references the same set of foods that "snack" does?

snack:
  - exact synonym: canape

OR we define canape as a kind of snack which is present at a formal occasion. Hypothetically, this could have logic like:

`canape: equivalent to: subClassOf snack and 'located in' some 'contains process' some 'formal event'`

where formal event is a class of process containing formal party, reception etc.

In the above diagram it looks like we would try to define "midnight snack" as a snack which is input to a "nighttime meal" process?
etc.

**Comment by ddooley** [comment]
We have started a separate thread https://github.com/FoodOntology/foodon/issues/271 to discuss analysis of when to precompose terms or not.  Regarding this thread, I think we'd precompose terms for cuisine around the world by region or culture, but we wouldn't precompose cuisine x meal type to avoid a combination bomb!

**Comment by ddooley** [comment]
Re. meal as a kind of food consumption process, we need help defining (and researching related definitions in OBO Foundry) for:

**food ingestion**: ~ having food arrive inside digestive tract (as opposed to intravenous feeding?)
vs 

**food consumption**: physical (aka teach a robot) act of getting food into body (mastication, salivation, + dexterity ...?)

Is one a component of the other?

**Comment by ddooley** [comment]
Here's a revision that seeks to simplify things a bit.  Not all the way there, but closer... I don't think we would add the "Needed in research?" items unless requested to do so.  A good writeup on meal courses: https://www.webstaurantstore.com/blog/2578/full-course-meal.html  

<img width="1050" alt="image" src="https://github.com/user-attachments/assets/eee3ca2d-ab7f-45fc-989f-ffe3a494d6ff">
